cmd/query/app/server.go

```diff
@@ -123,7 +127,9 @@ func (s *Server) Start() error {
 	// Start cmux server concurrently.
 	go func() {
 		s.svc.Logger.Info("Starting CMUX server", zap.Int("port", s.queryOptions.Port))
-		if err := cmuxServer.Serve(); err != nil {
+
+		err := cmuxServer.Serve()
+		if err != nil && !strings.Contains(err.Error(), "use of closed network connection") {
```

Comment:
strcmp is unfortunate. I see that cmux has ErrListenerClosed, at least (https://github.com/soheilhy/cmux/blob/master/cmux.go#L62). Is there no cleaner way to check for server closed? If not, maybe open a ticket w/ cmux?

Reply:
Agree with you :) There is no nice way to check for this currently, also cmux doesn't have nice way to close server. I opened this PR with cmux just now https://github.com/soheilhy/cmux/pull/69, this is still WIP. Until then, should be not fix this part? Or i could add a TODO about this cmux PR

Reply:
Once that cmux PR is merged, the idea is to just call `cmux.Close()` and no such error checking needed

Reply:
+1

Reply:
well, you'd still need to check the error to detect normal exit, because server.Close() could be called from the interrupt.